pkg/apis/controlplane/v1alpha1/types.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+// The category of NetworkPolicy, could be NetworkPolicy, ClusterNetworkPolicy, and AntreaNetworkPolicy.
+type NetworkPolicyCategory string
+
+const (
+	// It represents the K8s NetworkPolicy.
+	NetworkPolicy NetworkPolicyCategory = "NetworkPolicy"
+	// It represents the Antrea ClusterNetworkPolicy.
+	ClusterNetworkPolicy NetworkPolicyCategory = "ClusterNetworkPolicy"
+	// It represents the Antrea NetworkPolicy.
+	AntreaNetworkPolicy NetworkPolicyCategory = "AntreaNetworkPolicy"
+)
+
+// NetworkPolicyReference identifies a NetworkPolicy across all kinds of NetworkPolicies.
+type NetworkPolicyReference struct {
+	// The category of the NetworkPolicy.
+	Category NetworkPolicyCategory `json:"category,omitempty"`
+	// The Namespace of the NetworkPolicy. It's empty for ClusterNetworkPolicy.
+	Namespace string `json:"namespace,omitempty"`
+	// The name of the NetworkPolicy.
+	Name string `json:"name,omitempty"`
+}
+
+// Stats contains several stats of a NetworkPolicy.
+type Stats struct {
+	// Packets is the packets count.
+	Packets int64 `json:"packets,omitempty"`
+	// Bytes is the bytes count.
+	Bytes int64 `json:"bytes,omitempty"`
+	// Sessions is the sessions count.
+	Sessions int64 `json:"sessions,omitempty"`
+}
+
+// NetworkPolicyStats contains the information and stats of a NetworkPolicy.
+type NetworkPolicyStats struct {
+	// The reference of the NetworkPolicy.
+	NetworkPolicy NetworkPolicyReference `json:"networkPolicy,omitempty"`
+	// The stats of the NetworkPolicy.
+	Stats Stats `json:"stats,omitempty"`
+}
+
+// NodeSummary contains a list of NetworkPolicyStats and the information about the Node.
+// It's the struct that will be reported to the antrea-controller by the antrea-agent.
+type NodeSummary struct {
```

Comment:
Will you use the struct for other purposes later? If not, we should call it NodeStatsSummary?